File: src/main/java/recursion/Palindrome.java
package recursion;

import java.util.logging.Logger;

public class Palindrome {

    public static void main(String[] args) {
        Logger lg = Logger.getLogger("Palindrome");
        String str = "Bob";
        lg.info("is " + str + " palindrome ? " + checkPalindrome(str.toCharArray(), 0, str.length()-1));

    }

    private static boolean checkPalindrome(char[] cArr, int startIdx, int endIdx) {
        if(startIdx>=endIdx) {
            return true;
        }
        return cArr[startIdx]==cArr[endIdx] && checkPalindrome(cArr, ++startIdx, --endIdx);
    }
}

File: src/main/kotlin/recursion/Palindrome.kt
package recursion

import java.util.logging.Logger

fun main() {
    val lg = Logger.getLogger("Palindrome")
    val str = "Bob"
    lg.info("is ${str} palindrome ? : ${checkPalindrome(str.toCharArray(), 0, str.length-1)}")
}

private fun checkPalindrome(str: CharArray, strIdx: Int, endIdx: Int): Boolean {
    if(strIdx>=endIdx){
        return true
    }
    return str[strIdx]==str[endIdx] && checkPalindrome(str, strIdx+1, endIdx-1)

}
